From c455ed66ae059ce92500558926f5aa15c91ea772 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Tue, 31 Aug 2021 16:38:48 +0000
Subject: IWYU: add string.h for strlen in text_fragments_utils

Remove namespace std from strlen to avoid including cstring as well.

Bug: None
Change-Id: I234e5d0a96ecf18342ebe322d011914477d75afb
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3132943
Reviewed-by: Tommy Martino <tmartino@chromium.org>
Commit-Queue: Stephan Hartmann <stha09@googlemail.com>
Cr-Commit-Position: refs/heads/main@{#916839}
---
 .../core/common/text_fragments_utils.cc                    | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/components/shared_highlighting/core/common/text_fragments_utils.cc b/components/shared_highlighting/core/common/text_fragments_utils.cc
index 9f39ce69e7af1..013c364ec003c 100644
--- a/components/shared_highlighting/core/common/text_fragments_utils.cc
+++ b/components/shared_highlighting/core/common/text_fragments_utils.cc
@@ -4,6 +4,8 @@
 
 #include "components/shared_highlighting/core/common/text_fragments_utils.h"
 
+#include <string.h>
+
 #include <sstream>
 
 #include "base/json/json_writer.h"
@@ -53,9 +55,8 @@ bool SplitUrlTextFragmentDirective(const std::string& full_url,
   *webpage_url = GURL(full_url.substr(0, pos - 1));
 
   // We only want to keep what's after the delimiter.
-  *highlight_directive =
-      full_url.substr(pos + std::strlen(kFragmentsUrlDelimiter) +
-                      std::strlen(kFragmentParameterName));
+  *highlight_directive = full_url.substr(pos + strlen(kFragmentsUrlDelimiter) +
+                                         strlen(kFragmentParameterName));
   return true;
 }
 
-- 
2.30.2

